sstables/writer.hh

```diff
@@ -42,12 +42,22 @@ class metadata_collector;
 class file_writer {
     output_stream<char> _out;
     writer_offset_tracker _offset;
+    std::string _filename;
 public:
-    file_writer(file f, file_output_stream_options options)
-        : _out(make_file_output_stream(std::move(f), std::move(options))) {}
+    file_writer(file f, file_output_stream_options options, std::string filename)
+        : _out(make_file_output_stream(std::move(f), std::move(options)))
+        , _filename(std::move(filename))
+    {}
+
+    file_writer(output_stream<char>&& out, std::string filename)
+        : _out(std::move(out))
+        , _filename(std::move(filename))
+    {}
 
     file_writer(output_stream<char>&& out)
-        : _out(std::move(out)) {}
+        : _out(std::move(out))
+        , _filename("output_stream")
```

Comment:
IMO passing "output_stream" as the file name is confusing, since it's not the actual file name which is used here. The file name is not known, so _filename should be a disengaged std::optional<> in this case.

Reply:
Using std::optional is similar to a previous implementation that was changed to this one based on a review comment...
When disengaged, what do you think should the printout say?
How about `<output_stream>` to make it more obvious it's not an actual file name?